server/controllers/entryController.js

```diff
@@ -46,7 +46,10 @@ class entryController {
     if (!entry) {
       return res.status(404).json({ status: 404, error: 'Entry not found' });
     }
-    return res.status(200).json({ status: 200, data: entry });
+    if (entry.userId === parseInt(req.user.id)) {

```

Comment:
Since you are going to use this permission check for PATCH, DELETE, GET, make this a middleware. You can call the middleware `permissions` or any name of your choice.

Reply:
implement the middleware in the endpoints were it's needed. That way you won't have repeated codes in your codebase

Reply:
Done